File: pbgestfinbowcl/src/app/gestione-contestazioni/services/contestazioni.service.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

import { HttpClient, HttpParams,HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from "rxjs";
import { ConfigService } from 'src/app/core/services/config.service';
import { AllegatiContestazioniVO } from '../commons/dto/allegatiContestazione';
import { ContestazioniVO } from '../commons/dto/contestazioni';


@Injectable({
  providedIn: 'root'
})
export class ContestazioniService {

  constructor(
    private configService: ConfigService,
    private http: HttpClient) { }


   getContestazioni(idProgetto) : Observable <Array<ContestazioniVO>>  {
      let url = this.configService.getApiURL() + '/restfacade/contestazioni/getContestazioni';
      let params = new HttpParams().set('idProgetto', idProgetto);
      return this.http.get<Array<ContestazioniVO>>(url, { params: params });
  } 


  getCodiceProgetto(idProgetto) {
    let url = this.configService.getApiURL() + '/restfacade/contestazioni/getCodiceProgetto';
    let params = new HttpParams().set('idProgetto', idProgetto);
    return this.http.get<any[]>(url, { params: params });
} 



 /*  getRecupProvRevoca(idProgetto){
    let url = this.configService.getApiURL() + '/restfacade/contestazioni/getRecupProvRevoca';
      let params = new HttpParams().set('idProgetto', idProgetto);
      return this.http.get<any[]>(url, { params: params });
  } */

  


  getAllegati(idContestazione) {
    let url = this.configService.getApiURL() + '/restfacade/contestazioni/getAllegati';
    let params = new HttpParams().set('idContestazione', idContestazione.toString());
    return this.http.get<any[]>(url, { params: params });
  }

 
 deleteAllegato(idFileEntita:number){
  let url = this.configService.getApiURL() + `/restfacade/contestazioni/deleteAllegato/${idFileEntita}`;
  return this.http.post<any>(url,idFileEntita)
  
}


  inserisciLetteraAllegato(datiBackend:AllegatiContestazioniVO[],idContestazione:number){
    let url = this.configService.getApiURL() + `/restfacade/contestazioni/inserisciLetteraAllegato/${idContestazione}`;
    return this.http.post<any>(url,datiBackend)
    
  }

  inserisciContestazione(idGestioneRevoca:number){
    let url = this.configService.getApiURL() + `/restfacade/contestazioni/inserisciContestazione/${idGestioneRevoca}`;
    return this.http.post<any>(url,idGestioneRevoca)
    
  }


 /*  aggiornaContestazione(datiBackend){
    let url = this.configService.getApiURL() + '/restfacade/contestazioni/aggiornaContestazione';
    let headers = new HttpHeaders().set('Content-Type', 'application/json');
    return this.http.post<any>(url,datiBackend,{headers})
    
  } */

  inviaContestazione(idContestazione:number){
    let url = this.configService.getApiURL() + `/restfacade/contestazioni/inviaContestazione/${idContestazione}`;
    return this.http.post<any>(url,idContestazione)
    
  }

  

 

  eliminaContestazione(idContestazione:number){
    let url = this.configService.getApiURL() + `/restfacade/contestazioni/eliminaContestazione/${idContestazione}`;
    return this.http.post<any>(url,idContestazione)
    
  }

 
}

File: pbgestfinbo/pbgestfinbo-jar/src/main/java/it/csi/pbandi/pbgestfinbo/business/service/ContestazioniService.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbgestfinbo.business.service;

import it.csi.pbandi.pbgestfinbo.integration.vo.AllegatiContestazioniVO;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.*;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;


@Path("/contestazioni")
public interface ContestazioniService {


    @GET
    @Path("/getContestazioni")
    @Produces(MediaType.APPLICATION_JSON)
    Response getContestazioni(
            @DefaultValue("") @QueryParam("idProgetto") Long idProgetto,
            @Context HttpServletRequest req);

    @GET
    @Path("/getCodiceProgetto")
    @Produces(MediaType.APPLICATION_JSON)
    Response getCodiceProgetto(
            @DefaultValue("") @QueryParam("idProgetto") Long idProgetto,
            @Context HttpServletRequest req);

    @GET
    @Path("/getAllegati")
    @Produces(MediaType.APPLICATION_JSON)
    Response getAllegati(
            @DefaultValue("0") @QueryParam("idContestazione") Long idContestazione,
            @Context HttpServletRequest req);

    @POST
    @Path("/inserisciLetteraAllegato/{idContestazione}")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
	Response inserisciLetteraAllegato(
            @PathParam("idContestazione") Long idContestazione,
            List<AllegatiContestazioniVO> allegati,
            @Context HttpServletRequest req);

    @POST
    @Path("/deleteAllegato/{idFileEntita}")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Response deleteAllegato(
            @PathParam("idFileEntita") Long idFileEntita,
            @Context HttpServletRequest req
    );

    @POST
    @Path("/inserisciContestazione/{idGestioneRevoca}")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
	Response inserisciContestazione(
            @PathParam("idGestioneRevoca") Long idGestioneRevoca,
			@Context HttpServletRequest req);

    @POST
    @Path("/inviaContestazione/{idContestazione}")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
	Response inviaContestazione(
            @PathParam("idContestazione") Long idContestazione,
			@Context HttpServletRequest req);

    @POST
    @Path("/eliminaContestazione/{idContestazione}")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
	Response eliminaContestazione(
            @PathParam("idContestazione") Long idContestazione,
			@Context HttpServletRequest req);

}

